src/views/class/ProblemSetGrades.vue

```diff
@@ -0,0 +1,188 @@
+        draggingMap[col.dataIndex] = null
+      })
+      const draggingState = Vue.observable(draggingMap)
+      return ResizableTableHeader(h, props, children, columns, draggingState)
+    }
+    return {
+      problemSetID: this.$route.params.problemSetID,
+      classID: this.$route.params.classID,
+      now_user_id: this.$store.state.user.info.id,
+      problemSet: {}, // 初始化
+      columns: columns,
+      components: {
+        header: {
+          cell: ResizableTableHeaderWithColumns
+        }
+      },
+      problem_set: {
+        name: '',
+        description: '',
+        time: [moment(), moment()],
+        problems: []
+      },
+      title: '',
+      json_fields: {
+        学号: 'user.username',
+        姓名: 'user.nickname'
+      },
+      DetailsForm: [],
+      deleting: {},
+      deleteModelText: '',
+      visible: false,
+      confirmLoading: false,
+      data: [],
+      pagination: {
+        showTotal: (total, range) => `共 ${total} 条数据`
+      },
+      loading: false,
+      sorter: {
+        sortField: 'user.username',
+        sortOrder: 'ascend'
+      }
+    }
+  },
+  mounted () {
+    console.log(this.columns)
+    // console.log(this.$route.params)
+    getProblemSetGrades(this.classID, this.problemSetID).then(resp => {
+      console.log(resp)
+      const dynamicColumn = resp.problem_set.problems.map(item => {
+        return {
+          title: `${item.id} ${item.name}`,
+          dataIndex: item.id,
+          width: 180
+        }
+      })
+      for (let i = 0; i < dynamicColumn.length; i++) {
+        const _index = i + 2
+        this.columns.splice(_index, 0, dynamicColumn[i])
+      }
+      this.problemSet = resp.problem_set
+      this.title = `${resp.problem_set.name}成绩表格`
+      this.data = resp.problem_set.grades.map(item => {
+        const _detail = JSON.parse(item.detail)
+        // 对于每个问题ID，如果没有对应的值，则将值置为0
+        for (const problem of resp.problem_set.problems) {
+          const problemID = problem.id.toString()
+          if (!(_detail.hasOwnProperty(problemID))) {
+            _detail[problemID] = 0 // 如果没有对应的值，则将值置为0
+          }
+        }
+        return {
+          ...item,
+          ..._detail
+        }
+      })
+      this.DetailsForm = this.data
+      this.DetailsForm.sort((a, b) => a.user.username - b.user.username)
+    }).catch(err => {
+      this.$error({
+        title: '发生错误',
+        content: err.response.message + err.response.error
+      })
+    })
+  },
+  watch: {
+    '$route' (r) {
+      this.search_user_id = this.$route.query && +this.$route.query.user_id || null
+    }
+  },
+  methods: {
+    format (time) {
+      return moment(time).format('lll')
+    },
+    populateJsonFields () {
+      this.problemSet.problems.forEach((problem) => {
+        // 将问题的名称作为键，问题的ID作为值，添加到 json_fields 中
+        this.$set(this.json_fields, problem.name, problem.id.toString())
+      })
+      this.$set(this.json_fields, '总分', 'total')
+    },
```

Comment:
Why do we need this function? Can't this be done in line 156?

Reply:
![image](https://github.com/EduOJ/frontend/assets/126382401/4a736a41-51de-43a2-bd1a-d963cdaf8c70)
Line 156 provides form data, while this function provides column names.

Reply:
Yes, but can't we merge this function to line 156?